lib/library.rb

```diff
@@ -0,0 +1,40 @@
+require 'date'
+
+class Library
+  attr_accessor :items
+
+  def initialize(options={})
+    @items = options[:items] || []
+  end
+
+  def check_out(item, user)
+    @items.detect do |i|
+      if i[:item][:title] == item[:item][:title]
+        i[:available] = false
+        i[:return_date] = set_return_date
+      end
+    end
+    user.book_shelf.push check_out_message(item)
+  end
+
+  def find(args={})
+    @items.detect{|obj| obj[:item][args.keys.first] == (args[args.keys.first]).to_s }
```

Comment:
Metrics/LineLength: Line is too long. [85/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)